modules/nf-core/phispy/tests/main.nf.test

```diff
@@ -0,0 +1,69 @@
+// nf-core modules test phispy
+nextflow_process {
+
+    name "Test Process PHISPY"
+    script "../main.nf"
+    process "PHISPY"
+
+    tag "modules"
+    tag "modules_nfcore"
+    tag "phispy"
+
+    test("bacteroides_fragilis - gbff") {
+
+        when {
+            process {
+                """
+                input[0] = [
+                    [ id:'test' ], // meta map
+                    file( params.modules_testdata_base_path + "/genomics/prokaryotes/bacteroides_fragilis/genome/genome.gbff.gz", checkIfExists: true )
+                ]
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert path(process.out.gbk[0][1]).linesGzip.toString().contains('5150703 bp') },
+                { assert snapshot(process.out.versions).match() }
+            )
+            with(process.out.log[0]) {
+                assert path(get(1)).text.contains('NZ_CP069563.1')
+                assert path(get(1)).text.contains('91')
+                assert path(get(1)).text.contains('Kept')
+            }
+            with(process.out.coordinates[0]) {
+                assert path(get(1)).text.contains('NZ_CP069563.1')
+                assert path(get(1)).text.contains('pp1')
+                assert path(get(1)).text.contains('Longest Repeat flanking phage and within 2000 bp')
+            }
+        }
+
+    }
+
+    test("bacteroides_fragilis - gbff - stub") {
+
+        options "-stub"
+
+        when {
+            process {
+                """
+                input[0] = [
+                    [ id:'test' ], // meta map
+                    file( params.modules_testdata_base_path + "/genomics/prokaryotes/bacteroides_fragilis/genome/genome.gbff.gz", checkIfExists: true )
+                ]
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(process.out.versions).match() }
```

Comment:
```suggestion
                { assert snapshot(process.out).match() }
```

Reply:
The snapshot isn't stable for the stub (not sure why), that's why I'm testing just the versions snapshot.

Reply:
It's throwing an EOFException
```
│ java.io.EOFException                                                                                                                                                                        │
│     at java.base/java.util.zip.GZIPInputStream.readUByte(GZIPInputStream.java:268)                                                                                                          │
│     at java.base/java.util.zip.GZIPInputStream.readUShort(GZIPInputStream.java:258)                                                                                                         │
│     at java.base/java.util.zip.GZIPInputStream.readHeader(GZIPInputStream.java:164)                                                                                                         │
│     at java.base/java.util.zip.GZIPInputStream.<init>(GZIPInputStream.java:79)                                                                                                              │
│     at java.base/java.util.zip.GZIPInputStream.<init>(GZIPInputStream.java:91)                                                                                                              │
│     at com.askimed.nf.test.util.FileUtil.getMd5(FileUtil.java:111)                                                                                                                          │
│     at com.askimed.nf.test.lang.extensions.PathExtension.getMd5(PathExtension.java:21)                                                                                                      │
```
I wonder if it's because it's not receiving a proper gzipped file since your extension is based on your input. I think if you just test the gzipped file exists, the rest should be stable.

Reply:
I just did a test and it seems to be exactly what I thought. I put this ugly thing into the stub
```
    ${gbk_extension.endsWith('.gz')? "gzip -c ${prefix}.tsv > ${prefix}${gbk_extension}": "touch ${prefix}${gbk_extension}"}

```
and snapshoting everything works fine.

Reply:
Ah I see, that seems to be the case yes. In [8559855](https://github.com/nf-core/modules/pull/5292/commits/8559855de1bfe29b0c64089487e11ccc62791f61) I'm checking the snapshot of everything else except `.gbk` and also checking if `.gbk` exists, let me know if there's a more elegant way of doing it.